js/gdm/loginDialog.js

```diff
@@ -800,19 +932,29 @@ const LoginDialog = new Lang.Class({
 
         this._promptEntry.grab_key_focus();
 
-        let hold = new Batch.Hold();
+        this._holdForAnswer = new Batch.Hold();
         let tasks = [function() {
-                         this._prepareDialog(forSecret, hold);
+                         this._prepareDialog(forSecret);
                      },
 
-                     hold];
+                     this._holdForAnswer];
 
         let batch = new Batch.ConcurrentBatch(this, tasks);
 
         return batch.run();
     },
 
-    _prepareDialog: function(forSecret, hold) {
+    _onPromptEntryTextChanged: function() {
+        if (this._passwordResetCode == null) {
+            this._updateSignInButtonSensitivity(this._promptEntry.text.length > 0);
+        } else {
+            this._updateSignInButtonSensitivity(
+                this._promptEntry.text.length == _RESET_CODE_LENGTH &&
+                this._promptEntry.text.search(/\D/) == -1);
```

Comment:
You should be able to enter any password with one or more characters, but you should only be able to enter a password unlock code with exactly `_RESET_CODE_LENGTH` characters. If `this._passwordResetCode != null` then the user requested a password reset and needs to enter the unlock code.